tests/Unit/ProtoTests/Basic/out/src/Client/BasicClient.php

```diff
@@ -144,6 +146,7 @@ public function __construct(array $options = [])
     {
         $clientOptions = $this->buildClientOptions($options);
         $this->setClientOptions($clientOptions);
+        $options = array_merge($options, $this->setEmulatorConfig($options));
```

Comment:
We should merge the emulator config options in _before_ calling `setClientOptions`. I would even suggest doing it before `buildClientOptions`:
```suggestion
        $clientOptions = $this->buildClientOptions($options + $this->getDefaultEmulatorConfig()); 
        $this->setClientOptions($clientOptions);
```

Reply:
Sounds good